hudi-utilities/src/main/java/org/apache/hudi/utilities/HoodieSnapshotExporter.java

```diff
@@ -47,36 +49,47 @@
 import org.apache.spark.sql.SparkSession;
 import org.apache.spark.sql.execution.datasources.DataSource;
 
-import scala.Tuple2;
-import scala.collection.JavaConversions;
-
 import java.io.IOException;
 import java.io.Serializable;
 import java.util.ArrayList;
-import java.util.Arrays;
 import java.util.List;
 import java.util.stream.Collectors;
 
+import scala.Tuple2;
+import scala.collection.JavaConversions;
+
 /**
  * Export the latest records of Hudi dataset to a set of external files (e.g., plain parquet files).
  *
  * @experimental This export is an experimental tool. If you want to export hudi to hudi, please use HoodieSnapshotCopier.
  */
 public class HoodieSnapshotExporter {
+
+  @FunctionalInterface
+  public interface Partitioner {
+
+    DataFrameWriter<Row> partition(Dataset<Row> source);
+
+  }
```

Comment:
Also need UT for user-defined Partitioner?

Reply:
Sure. Will fix this once #1404 merged.